pkg/test/main/main.go

```diff
@@ -243,6 +274,21 @@ func main() {
 		}
 	}
 
+	if pprof_enabled {
+
+		for _, prof := range []string{"block", "goroutine", "mutex"} {
+			p := pprof.Lookup(prof)
+			filePath := fmt.Sprintf("%s_profile_%s.pb.gz", prof, mode)
+			log.Printf("storing %s profile for %s in %s", prof, mode, filePath)
+			f, err := os.Create(filePath)
```

Comment:
Do you think it makes sense to have multiple files created out of this? I'm not entirely sure how pprof generates its artifacts but would it be possible to have everything written to one file?

Reply:
The [pprof](https://github.com/google/pprof) tool really wants this. If this gets used we might do some intermediate software to manage files.